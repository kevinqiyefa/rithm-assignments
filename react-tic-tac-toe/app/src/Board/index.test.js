import React from 'react'; // for JSX
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import Board from '.'; // import the component itself

describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(
      <Board
        board={[[null, null, null], [null, null, null], [null, null, null]]}
        frozen={false}
        takeTurn={jest.fn()}
      />
    );
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
