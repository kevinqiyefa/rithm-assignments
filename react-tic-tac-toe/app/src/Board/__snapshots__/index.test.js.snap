// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
Array [
  Array [
    <Square
      handleClick={[Function]}
      id="sq00"
      key="0|0"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq01"
      key="0|1"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq02"
      key="0|2"
      val={null}
/>,
  ],
  Array [
    <Square
      handleClick={[Function]}
      id="sq10"
      key="1|0"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq11"
      key="1|1"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq12"
      key="1|2"
      val={null}
/>,
  ],
  Array [
    <Square
      handleClick={[Function]}
      id="sq20"
      key="2|0"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq21"
      key="2|1"
      val={null}
/>,
    <Square
      handleClick={[Function]}
      id="sq22"
      key="2|2"
      val={null}
/>,
  ],
]
`;
